cli/defenseclaw/commands/cmd_setup.py

```
@@ -2731,19 +3116,62 @@ def _rotate_token_transaction(
     # this on-disk A/B boundary: stop loads config+dotenv A, while start reloads
     # config+dotenv B under the explicit data directory. Do not reuse a
     # post-commit loader result for the authenticated stop phase.
-    with locked_config_yaml(config_file), locked_file_update(dotenv_path):
+    requested_scopes = sorted({_rotate_token_hook_scope(name) for name in scoped_connectors})
+    with locked_config_yaml(config_file), locked_file_update(dotenv_path), ExitStack() as hook_locks:
         config_snapshot, config_mode = _snapshot_regular_file(config_file, what="gateway config")
         authoritative_cfg = load_config(data_dir=data_dir)
         current_config, current_mode = _snapshot_regular_file(config_file, what="gateway config")
         if current_config != config_snapshot or current_mode != config_mode:
             raise click.ClickException("Gateway configuration changed while token rotation was taking its snapshot.")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)